fsspec/implementations/cached.py

```diff
@@ -868,15 +888,18 @@ def discard(self):
 
     def commit(self):
         self.fs.put(self.fn, self.path, **self.kwargs)
-        try:
-            os.remove(self.fn)
-        except (PermissionError, FileNotFoundError):
-            # file path may be held by new version of the file on windows
-            pass
+        # we do not delete local copy - it's still in the cache
 
     @property
     def name(self):
         return self.fn
 
+    def __repr__(self) -> str:
+        return f"LocalTempFile: {self.path}"
+
     def __getattr__(self, item):
-        return getattr(self.fh, item)
+        try:
+            return getattr(self.fh, item)
+        except AttributeError:
+            pass
+        raise AttributeError(item)
```

Comment:
why not use a `raise from` here to reraise AttributeError and not through away in the info of the previous attribute error?

Reply:
It will look like it failed from this object, but it will give additional information about it? It will just additional information. I'm worried about AttributeError information being deleted here.

Reply:
fine...